### PR TITLE
[Modules] Update the AKS API Version to latest and added new features

### PR DESCRIPTION
**Pull request by ahmadabdalla**
# Description

Updated AKS API Version and added new features.

### Pipeline references

| Pipeline |
| - |
| [ContainerService - ManagedClusters](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerservice.managedclusters.yml)|

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/container-service/managed-cluster/main.bicep [hunk 1]
```diff
@@ -405,18 +459,14 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2023-05-02-p
...
   properties: {
+    httpProxyConfig: !empty(httpProxyConfig) ? httpProxyConfig : null
+    identityProfile: !empty(identityProfile) ? identityProfile : null
     diskEncryptionSetID: !empty(diskEncryptionSetID) ? diskEncryptionSetID : null
     kubernetesVersion: (empty(aksClusterKubernetesVersion) ? null : aksClusterKubernetesVersion)
     dnsPrefix: aksClusterDnsPrefix
     agentPoolProfiles: primaryAgentPoolProfile
     linuxProfile: (empty(aksClusterSshPublicKey) ? null : aksClusterLinuxProfile)
     servicePrincipalProfile: (empty(aksServicePrincipalProfile) ? null : aksServicePrincipalProfile)
-    ingressProfile: {
```
Comment (AlexanderSehr) [inline modules/container-service/managed-cluster/main.bicep:414]: Why was this deleted? 

I see we still have the `dnsZoneResourceId` input parameter and a role assignment on the bottom of the template. Yet the cluster doesn't know about the DNS Zone anymore :o

File: modules/container-service/managed-cluster/main.bicep [hunk 2]
```diff
@@ -465,7 +515,9 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2023-05-02-p
     nodeResourceGroup: nodeResourceGroup
     enablePodSecurityPolicy: enablePodSecurityPolicy
     networkProfile: {
+      networkDataplane: !empty(aksClusterNetworkDataplane) ? any(aksClusterNetworkDataplane) : null
```
Comment (AlexanderSehr) [inline modules/container-service/managed-cluster/main.bicep:518]: This comment may affect mulitple places: Should this not be just `networkDataplane` as per the provider? The `aksCluster` in the prefix seems fairly redundant and I assume you added it for consistency reasons as other parameters have it to. So the question is then if we should remove that prefix from all parameters in the template.

Comment (AlexanderSehr) [reply modules/container-service/managed-cluster/main.bicep]: Love it, thanks

File: modules/container-service/managed-cluster/main.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply modules/container-service/managed-cluster/main.bicep:414]: glitch in the matrix :D restored.

**Review by AlexanderSehr — APPROVED** [review]
